src/index.ts

```diff
@@ -138,3 +139,54 @@ export function useLiveAnchorAccount<
     slotUpdated,
   };
 }
+
+function fetcher<I extends Idl, A extends keyof IdlAccounts<I>>(
+  program: Program<I>,
+) {
+  return function (method: A, ...params: [Address]): IdlAccounts<I>[A] {
+    const address = params[0];
+    return program.account[method].fetch(address) as IdlAccounts<I>[A];
+  };
+}
+
+export interface UseSWRAnchorAccountResult<
+  I extends Idl,
+  A extends keyof IdlAccounts<I>,
+> {
+  loading: boolean;
+  account?: IdlAccounts<I>[A];
+  mutate: KeyedMutator<IdlAccounts<I>[A]>;
+  error?: string;
+  isValidating: boolean;
+}
+
+export function useSWRAnchorAccount<
+  I extends Idl,
+  A extends keyof IdlAccounts<I>,
+>(
+  program: Program<I> | null | undefined,
+  accountType: A,
+  address: Address | null | undefined,
+  swrOptions?: SWRConfiguration,
+): UseSWRAnchorAccountResult<I, A> {
+  const {
+    data: account,
+    error,
+    mutate,
+    isValidating,
+  } = useSWR<IdlAccounts<I>[A]>(
+    program && address && [accountType, address],
```

Comment:
```suggestion
    program && address && [accountType, translateAddress(address)]],
```

If you look in anchor the typedef for `Address` is `PublicKey | string` where string is the base-58 encoded representation of a `PublicKey`. So I think this will cause `"Stake11111111111111111111111111111111111111"` and `new PublicKey("Stake11111111111111111111111111111111111111")` to have different keys. 

`useSWR` is supposed to handle complex objects as keys now with 1.1.0 so having a `Publickey` as a key should be fine right?

We will also need to upgrade `anchor` to `^0.20.0` as well because thats the version that fixes `translateAddress`: https://github.com/project-serum/anchor/blob/master/CHANGELOG.md#fixes-3. Also all our programs are now ^0.20.0 anyway

Reply:
Good catch. Do I need to update the `anchor` package first?

Reply:
yeah we need to update anchor. From the looks of the changelog, there aren't any on the typescript side (tho its another story for the rust on-chain programming side hahaha)